docs: add BentoCloud docs (#3984)
bentoml/BentoML: docs/source/bentocloud/getting-started/scale.rst

================
Scale for Growth
================

In the previous parts of our 'Getting Started' series:

> - :ref:`<build>`
> - :ref:`<ship>`

Reviewer: ```suggestion
* :doc:`/bentocloud/getting-started/build`
* :doc:`/bentocloud/getting-started/ship`
* :doc:`/bentocloud/getting-started/scale`
```

> - :doc:`Getting Started -- Build AI Application <build>`
> - :doc:`Getting Started -- Ship To Production <ship>`

Reviewer: ```suggestion
- :ref:`<build>`
- :ref:`<ship>`
```

================
Scale for Growth
================

In the previous parts of our 'Getting Started' series:

- :doc:`Getting Started -- Build AI Application <build>`
- :doc:`Getting Started -- Ship To Production <ship>`

> We've shown you how to build your application with BentoML and deploy it using BentoCloud. Now, in this final article, we'll explain how BentoCloud makes scaling your application straightforward. BentoCloud is a fully managed platform that aims to relieve users of hardware concerns, allowing you to pay only for what you use.

Reviewer: ```suggestion
We've shown you how to build your application with BentoML and deploy it using BentoCloud. Now, in this final article, we'll explain how BentoCloud makes scaling your application straightforward. BentoCloud is a fully managed platform that aims to relieve users of infrastructure concerns, allowing you to pay only for what you use.
```

-------------------
Resource Management
-------------------

~~~~~~~~~~~~~~~~~~~~~~
Auto Scaling Instances
~~~~~~~~~~~~~~~~~~~~~~

> BentoCloud is built distributedly, taking full advantage of the auto scaling capabilities provided by its  `Horizontal Pod Autoscaler <https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/>`_ (HPA). 
> We've set up a range of metrics that trigger the autoscaler, while also giving expert users the freedom to override these default scaling policies. This can be done via a JSON editor or directly through the BentoCloud API.

Reviewer: ```suggestion
BentoCloud is designed with scalability at its core. Scaling operations can be performed based on various parameters including CPU or GPU usage, or metrics related to traffic volume. The platform's advanced configuration settings enable a wide range of scaling behaviors, accommodating needs from rapid scaling for high-growth scenarios to stable scaling for consistent demand levels. Importantly, BentoCloud operates on a pay-as-you-go model, ensuring users are only charged for the resources they have actually utilized, promoting cost-efficiency and flexibility.
```

.. image:: ../../_static/img/bentocloud/scale-autoscale.png

For those seeking advanced scaling behaviour, your services can scale according to `custom metrics <https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#scaling-on-custom-metrics>`_. 
This feature is exclusively available to users with **Enterprise** plan.

~~~~~~~~~~
Scale-To-Zero
~~~~~~~~~~

> When you deploy a Bento, you can opt for an on-demand function mode of deployment. 
> This is BentoML's serverless approach, ideally suited for use cases where ML models are large, such as high-resolution image models or large language models. 
> Instances can scale down to zero when not in use, offering a cost-effective solution, especially when using pricier GPU instances.

Reviewer: ```suggestion
BentoCloud supports deploying applications as on-demand functions, enhancing its flexibility and resource efficiency. In this deployment mode, instances can scale down to zero when there's no usage on the API server or runners, minimizing idle resources. To ensure high reliability during a cold start, BentoCloud queues requests, preventing potential overloads. This deployment strategy is particularly advantageous for applications handling a large number of models and sporadic requests, as only those models actively in use are allocated resources. Additionally, this mode of operation provides significant cost efficiency, especially for GPU-accelerated instances, by tailoring resource allocation to real-time demand.
```
